api/src/modules/h3-data/h3-data.repository.ts

```diff
@@ -30,4 +30,49 @@ export class H3DataRepository extends Repository<H3Data> {
       );
     }
   }
+
+  /** Retrieves single crop data by a given resolution
+   *
+   * @param h3Id: Name of the column inside the dynamically generated table
+   * @param resolution: An interger between 1 (min resolution) and 6 (max resolution).
+   * Resolution validation done at route handler
+   *
+   * @returns: Single object with the h3Index as property key, and its value
+   *
+   * @debt: This should be done on a single raw sql query, but I couldn't make it work
+   * Super Senior powers requested
+   */
+
+  async getH3ByIdAndResolution(
+    h3Id: string,
+    resolution: number,
+  ): Promise<H3IndexValueData> {
+    const h3Info = await this.createQueryBuilder('h3')
+      .select('"h3tableName"')
+      .addSelect('"h3columnName"')
+      .where('h3.id = :id', { id: h3Id })
+      .getRawOne();
+
+    if (!h3Info)
+      throw new NotFoundException(
+        `Requested H3 with ID: ${h3Id} could not been found`,
+      );
+    try {
+      const h3data = await this
+        .query(` select h3parent.h3_to_parent, sum(${h3Info.h3columnName})  from
```

Comment:
Couldn't this be simplified as :

`
            select h3_to_parent(h3index, ${resolution}) as h3_to_parent, sum(${h3Info.h3columnName})
            from ${h3Info.h3tableName} hgsvrgp
            group by h3_to_parent
`

Reply:
Yes, brilliant, we love KISS.
Thanks!